ReactSkia/core_modules/RSkImageLoader.cpp

```diff
@@ -0,0 +1,139 @@
+            getImageSize(jsArgAsString(args,0),resolveBlock,rejectBlock);
+          }),
+      Method(
+          "getSizeWithHeaders",
+          [this] (dynamic args, CxxModule::Callback resolveBlock, CxxModule::Callback rejectBlock) {
+            getImageSizeWithHeaders(jsArgAsString(args, 0), jsArgAsString(args, 1), resolveBlock, rejectBlock);
+          }),
+      Method(
+          "prefetchImage",
+          [this] (dynamic args, CxxModule::Callback resolveBlock, CxxModule::Callback rejectBlock) {
+            prefetchImage(jsArgAsString(args, 0), resolveBlock, rejectBlock);
+          }),
+      Method(
+          "queryCache",
+          [this] (dynamic args, CxxModule::Callback resolveBlock, CxxModule::Callback rejectBlock) {
+            queryCache(jsArgAsArray(args, 0), resolveBlock, rejectBlock);
+          }),
+  };
+}
+
+std::string RSkImageLoaderModule::getName() {
+  return "ImageLoader";
+}
+
+void RSkImageLoaderModule::getImageSize(std::string uri, CxxModule::Callback resolveBlock, CxxModule::Callback rejectBlock) {
+  auto sharedCurlNetworking = CurlNetworking::sharedCurlNetworking();
+  std::shared_ptr<CurlRequest> remoteCurlRequest = std::make_shared<CurlRequest>(nullptr,uri,0,"GET");
+  sk_sp<SkImage> imageData{nullptr};
+  auto completionCallback =  [this,remoteCurlRequest , resolveBlock , rejectBlock](void* curlresponseData,void *userdata)->bool {
+    CurlResponse *responseData =  (CurlResponse *)curlresponseData;
+    CurlRequest * curlRequest = (CurlRequest *) userdata;
+    decodedimageCacheData imageCacheData;
+    if(responseData && (responseData->responseBuffer!=nullptr) && (responseData->contentSize >0)) {
+      sk_sp<SkData> data = SkData::MakeWithCopy(responseData->responseBuffer,responseData->contentSize);
+      if (data){
+        sk_sp<SkImage> remoteImageData = SkImage::MakeFromEncoded(data);
+        RNS_LOG_DEBUG("Network response received success");
+
+        std::vector<dynamic> imageDimensions;
+        imageDimensions.push_back(folly::dynamic::array(remoteImageData->width(),remoteImageData->height()));
+        resolveBlock(imageDimensions);
+        
+        imageCacheData.imageData = remoteImageData;
+        imageCacheData.expiryTime = (SkTime::GetMSecs() + DEFAULT_MAX_CACHE_EXPIRY_TIME);//convert sec to milisecond 60 *1000
+        RSkImageCacheManager::getImageCacheManagerInstance()->imageDataInsertInCache(curlRequest->URL.c_str(), imageCacheData);
+
+        //Reset the lamda callback so that curlRequest shared pointer dereffered from the lamda
+        // and gets auto destructored after the completion callback.
+        remoteCurlRequest->curldelegator.CURLNetworkingCompletionCallback = nullptr;
+        return 0;
+      }
+    }   
+    RNS_LOG_DEBUG("Network response received error");
+    std::vector<dynamic> imageError;
+    imageError.push_back(folly::dynamic::array("Image Load failed"));
+    rejectBlock(imageError);
+
+    imageRequestList_.erase(curlRequest->URL.c_str());
```

Comment:
Erase is handled only got error completion call back, for success case is not handled yet